src/service/performance-impl.js

```diff
@@ -590,19 +590,21 @@ export class Performance {
    *     when adding a new metric.
    * @param {number=} opt_delta The delta. Call tickDelta instead of setting
    *     this directly.
+   * @param {number=} opt_value The value to use. Overrides default calculation.
    */
-  tick(label, opt_delta) {
+  tick(label, opt_delta, opt_value) {
     const data = dict({'label': label});
     let delta;
 
-    // Absolute value case (not delta).
-    if (opt_delta == undefined) {
-      // Marking only makes sense for non-deltas.
+    if (opt_delta != undefined) {
```

Comment:
This creates a weird case where providing both optional params doesn't work correctly. We should either add a `devAssert` that both are not set, or we should remove the `opt_value` entirely. It seems that only `timeOrigin` is special, so we could move the "message or queue" code below to a private method, and call it with the necessary struct.

Reply:
You are right, `timeOrigin` is special. I've added the `devAssert`